### PR TITLE
Documentation fragment

### LINKED ISSUE #2715: Document java files which require documentation
### Description

The following java files need documentation:

- [ ] Common API Manager.java
- [ ] OpenFoodAPIClient.java
- [ ] WikidataApiClient.java
- [ ] CategoryRepository.java
- [ ] FileDownloader.java
- [ ] CategoryFragmentViewModel.java
- [ ] AdditiveFragmentHelper.java

_Please comment name of few more classes that needs documentation_

### PR DESCRIPTION
**Pull request by vanshikaarora**
## Description

Added Documentation for Category package

## Related issues and discussion
#fixes #2715 and #2719

### CODE REVIEW

**Comment by VaiTon** [comment]
For the entire thing:
* Use the `@return` tag in javadocs
* Format javadocs so that the opening line and the closing line are on their own line
* Check typos! 💯

**Comment by teolemon** [comment]
@vanshikaarora can you rebase ? If no update, I propose closing

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductIngredientsFragment.java [hunk 1]
```diff
@@ -201,11 +210,21 @@ public void onError() {
         }
     }
 
+    /**
+     * Set visibility parameters when image is loaded
+     * */
+
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductIngredientsFragment.java:216]:
```suggestion
    /**
     * Set visibility parameters when image is loaded
     * */
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductIngredientsFragment.java [hunk 2]
```diff
@@ -269,6 +288,10 @@ public void onError() {
         }
     }
 
+    /**
+     * Automaticcely load suggestions for allergen names
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductIngredientsFragment.java:292]:
```suggestion
     * Automatically load suggestions for allergen names
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductIngredientsFragment.java [hunk 3]
```diff
@@ -449,6 +482,12 @@ public void hideImageProgress(boolean errorInUploading, String message) {
         Toast.makeText(activity, message, Toast.LENGTH_SHORT).show();
     }
 
+    /**
+     * Display the list of ingredients based on the result from ocr of IngredientsList photo
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductIngredientsFragment.java:486]:
```suggestion
     * Displays the list of ingredients based on the result from OCR of IngredientsList photo
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductNutritionFactsFragment.java [hunk 4]
```diff
@@ -387,6 +394,11 @@ private void updateSelectedDataSize(String s) {
         radioGroup.jumpDrawablesToCurrentState();
     }
 
+    /**
+     * lads nutrition image into the ImageView
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductNutritionFactsFragment.java:398]:
```suggestion
     * loads nutrition image into the ImageView
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductNutritionFactsFragment.java [hunk 5]
```diff
@@ -720,6 +732,12 @@ public void getDetails(Map<String, String> targetMap) {
         }
     }
 
+    /**
+     * Add nutients to the map by from the text enetered into EditText
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductNutritionFactsFragment.java:736]:
```suggestion
     * Add nutrients to the map by from the text entered into EditText
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductNutritionFactsFragment.java [hunk 6]
```diff
@@ -720,6 +732,12 @@ public void getDetails(Map<String, String> targetMap) {
...
+     * @param editTextView EditText with spinner for entering the nutients
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductNutritionFactsFragment.java:737]:
```suggestion
     * @param editTextView EditText with spinner for entering the nutrients
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductOverviewFragment.java [hunk 7]
```diff
@@ -507,6 +516,10 @@ private void initializeChips() {
         }
     }
 
+    /**
+     * Auto load suggestions into various NachoTextViews
+     * */
+
     private void loadAutoSuggestions() {
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductOverviewFragment.java:523]:
```suggestion

    /**
     * Auto load suggestions into various NachoTextViews
     * */
    private void loadAutoSuggestions() {
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductOverviewFragment.java [hunk 8]
```diff
@@ -891,6 +909,10 @@ void selectProductLanguage() {
             .show();
     }
 
+    /**
+     * Before moving next check if the required feilds are empty
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductOverviewFragment.java:913]:
```suggestion
     * Before moving next check if the required fields are empty
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/HomeFragment.java [hunk 9]
```diff
@@ -192,6 +196,11 @@ public void onError(Throwable e) {
 
     }
 
+    /**
+     * Set text displayed on HOme based on build variant
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/HomeFragment.java:200]:
```suggestion
     * Set text displayed on Home based on build variant
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/HomeFragment.java [hunk 10]
```diff
@@ -192,6 +196,11 @@ public void onError(Throwable e) {
...
+     * @param totalProductCount count of total products availab;e on the apps database
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/HomeFragment.java:201]:
```suggestion
     * @param totalProductCount count of total products available on the apps database
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/PreferencesFragment.java [hunk 11]
```diff
@@ -256,6 +260,10 @@ public void onResume() {
         }
     }
 
+    /**
+     * AsyncTak to recieve additves on a background thread
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/PreferencesFragment.java:264]:
```suggestion
     * AsyncTask to receive additives on a background thread
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/CommonApiManager.java [hunk 12]
```diff
@@ -40,6 +44,10 @@ public ProductApiService getProductApiService() {
         return productApiService;
     }
 
+    /**
+     * Defines  and returns getOpenFoodApiService
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/CommonApiManager.java:48]:
```suggestion
     * Defines and returns getOpenFoodApiService
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/WikidataApiClient.java [hunk 13]
```diff
@@ -21,7 +21,7 @@
 import retrofit2.converter.jackson.JacksonConverterFactory;
 
 /**
- * Created by Shubham Vishwakarma on 14.03.18.
+ * API client to recieve data from WikiData APIs
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/WikidataApiClient.java:24]:
```suggestion
 * API client to recieve data from the Wikidata APIs
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/WikidataApiClient.java [hunk 14]
```diff
@@ -21,7 +21,7 @@
 import retrofit2.converter.jackson.JacksonConverterFactory;
 
 /**
- * Created by Shubham Vishwakarma on 14.03.18.
+ * API client to recieve data from the Wikidata APIs
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/WikidataApiClient.java:24]:
```suggestion
 * API client to receive data from the Wikidata APIs
```

**Pull request merged**